library/core/src/option.rs

```diff
@@ -707,7 +707,7 @@ impl<T> Option<T> {
     #[stable(feature = "pin", since = "1.33.0")]
     #[rustc_const_unstable(feature = "const_option_ext", issue = "91930")]
     pub const fn as_pin_ref(self: Pin<&Self>) -> Option<Pin<&T>> {
-        match Pin::get_ref(self).as_ref() {
+        match Pin::get_ref(self) {
             // SAFETY: `x` is guaranteed to be pinned because it comes from `self`
             // which is pinned.
             Some(x) => unsafe { Some(Pin::new_unchecked(x)) },
```

Comment:
Mixing coercions with unsafe like this makes me nervious. Have you tested this with miri?